test/yb-ctl-test.sh

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export YB_PG_FALLBACK_SYSTEM_USER_NAME=$USER
+export YB_DISABLE_CALLHOME=1
+
+# This will be auto-detected the first time yb-ctl auto-downloads and installs YugaByte DB.
+installation_dir=""
+
+log() {
+  echo >&2 "[$( date +%Y-%m-%dT%H:%M:%S )] $*"
+}
+
+detect_installation_dir() {
+  if [[ -z $installation_dir ]]; then
+    installation_dir=$( ls -td "$HOME/yugabyte-db/yugabyte-"* | head -1 )
+    log "YugaByte DB has been automatically installed into directory: $installation_dir"
+  fi
+}
+
+verify_ysqlsh() {
+  local node_to_connect=${1:-}
+  log "Creating a YSQL table and inserting a bit of data"
+  local ysqlsh_opts=""
+  if [[ -n $node_to_connect ]]; then
+    ysqlsh_opts="-h 127.0.0.$node_to_connect"
+  fi
+  "$installation_dir"/bin/ysqlsh $ysqlsh_opts <<-EOF
```

Comment:
do we default to 127.0.0.1 if we do not pass in anything? if yes, then the note you had below of why it doesn't work, is because we are not passing any args to this function, which would leave this arg empty and go on our default path...but since it's after you did `stop_node 1`, it is expected to fail, as that node 1 is the one on 127.0.0.1...

Reply:
Fixed.